tools/c7n_org/c7n_org/cli.py

```diff
@@ -583,6 +585,10 @@ def run_account(account, region, policies_config, output_path,
                 "Running policy:%s account:%s region:%s",
                 p.name, account['name'], region)
             try:
+                # azure needs to be initialized so we can use the correct cloud endpoints
+                if p.provider_name == 'azure' and region != 'global':
+                    p.options.regions = [region]
```

Comment:
Why do we need to explicitly initialize the provider here? Also, it will execute for each policy, so probably not the best place..

Reply:
When I was tracing the c7n-org execution, it never initialized the Azure provider on account run so the cloud rules never get set during the c7n-org workflow. I also needed to set the `regions` as the Azure provider's `_get_cloud_endpoints` function requires that option property to be set to find the correct cloud endpoints.

I'll try to see if I can find a better place to initialize this once instead of per policy and per account. We should really only need to initialize this once since users should only be running c7n-org against a single cloud. Any suggestions will be appreciated!

Reply:
Thoughts on placing it right after we create the config object within `run_account` function?

```
config = Config.empty(
    region=region, cache=cache_path,
    cache_period=cache_period, dryrun=dryrun, output_dir=output_path,
    account_id=account['account_id'], metrics_enabled=metrics,
    log_group=None, profile=None, external_id=None)

env_vars = account_tags(account)

# azure needs to be initialized so we can use the correct cloud endpoints
if account.get('provider') == 'azure' and region != 'global':
    config['regions'] = [region]
    Azure().initialize(config)
```

Reply:
Removed this code completely and have `get_session_factory` do the work